blog/2022/2022-04-06-loopback-4-maintianers-testimonial-rifa.md

```diff
@@ -0,0 +1,46 @@
+
+As basic HTTP and REST frameworks and libraries, they are well-suited for quickly buiding applications as developers need not worry about the nooks and crannies of developing web APIs. However, upholding this abstraction is simply not possible with every project, and eventually we would need to work at the protocol-level to cater to our business requirements or use a coding paradigm that was outside of the expectations of the framework developers. This meant digging through the underlying framework code to undertand its inner-workings or outright working around the system, effectively removing the nice abstractions that were built.
+
+> "Frameworks typically take an all-or-nothing approach - Either follow our application architecture style, or don't use it at all."
+
+This proved to be detremental to our projects, and meant that there was a level of unpredictability in the software development lifecycle in projects that implemented these frameworks; We were unable to fully account for the limitations of the frameworks, and were bogged down by the need to implement our own variation of certain fundamental abstractions that simpler libraries (such as Express) did not offer. This led to development time being unecessarily spent architecting and implementing solutions that did not solve the immediate business requirements which in turn led to code which required more maintenance, was not easily understood by different teams, and had incompatible approaches to solving similar issues. This led to code debt and time spent continuously reviewing and updating each project to synchronise them with new coding practices and paradigms that we may adopt. This compromise between ease of use and the need to architect each project ourselves proved challenging, and did not bode well for the longevity and maintainability of each project, putting them at an uneeded risk for a rewrite as the fast-paced Node.js ecosystem continues to develop new approaches to solving the same issues.
+
+## The "LoopBack Advantage"
+
+Hence when we discovered LoopBack 4, we were drawn and equally taken aback by the unorthodox approach it took; Like many others, we initially questioned the need for the significant amount of APIs, configuration files, and Node.js packages. As a project that angles itself as purely a REST framework, why was it visibly more complex? However, as we dug into how each package functions and the unique properties of the "LoopBack Approach", we realised we found a framework that had potential in solving the pain points of other projects. As we started building projects with LoopBack 4, we were initially frustrated and confused for the need of dependency injection and decorators; As the most of the Node.js ecosystem did not implement these features, we did not fully-understand the need for them and saw them as needless complexity. However, as we continued to build our projects, we soon realised the true strength of LoopBack 4.
```

Comment:
A bit abrupt, perhaps:

We were drawn and equally taken aback by the unorthodox approach LoopBack 4 took when we discovered it?